Modules/Filtering/Thresholding/include/itkHuangThresholdCalculator.hxx

```diff
@@ -102,7 +103,7 @@ HuangThresholdCalculator<THistogram, TOutput>::GenerateData()
   for (InstanceIdentifier threshold = m_FirstBin; threshold < m_LastBin; ++threshold)
   {
     double entropy = 0.;
-    auto   mu = Math::Round<MeasurementType>(W[threshold] / S[threshold]);
+    auto   mu = static_cast<MeasurementType>(std::round(W[threshold] / S[threshold]));
```

Comment:
What type is the results of `W[theshold]`? If it can be signed then this may change results in some cases.

Reply:
![h](https://github.com/InsightSoftwareConsortium/ITK/assets/1494580/e08c3cb0-9289-4b67-b5c6-5cf1fc6cc453)

Reply:
I don't think people would _expect_ negative measurement values of the form `-n.5` (for a natural number `n`) to be rounded towards zero. Do you?

Reply:
std::nearbyint (and std::rint) has 4 rounding modes:

```cpp
#include <iostream>
#include <cfenv>
#include <cmath>

namespace
{

void printrounding()
{
	std::cout << "\nRounding using ";
	switch (std::fegetround())
	{
	case FE_DOWNWARD:
		std::cout << "downward";
		break;
	case FE_TONEAREST:
		std::cout << "to-nearest";
		break;
	case FE_TOWARDZERO:
		std::cout << "toward-zero";
		break;
	case FE_UPWARD:
		std::cout << "upward";
		break;
	default:
		std::cout << "unknown";
		break;
	}
}

}

int main()
{
	// Default (FE_TONEAREST)
	printrounding();
	std::cout
		<< "\n -1.5 --> " << std::nearbyint(-1.5)
		<< "\n  1.5 --> " << std::nearbyint(1.5);

	// FE_TOWARDZERO
	std::fesetround(FE_TOWARDZERO);
	printrounding();
	std::cout
		<< "\n -1.5 --> " << std::nearbyint(-1.5)
		<< "\n  1.5 --> " << std::nearbyint(1.5);

	// FE_UPWARD
	std::fesetround(FE_UPWARD);
	printrounding();
	std::cout
		 << "\n -1.5 --> " << std::nearbyint(-1.5)
		 << "\n  1.5 --> " << std::nearbyint(1.5) << std::endl;

	return 0;
}
```

```
r@deb2:~/tmp$ g++ rounding.cpp 
r@deb2:~/tmp$ ./a.out 

Rounding using to-nearest
 -1.5 --> -2
  1.5 --> 2
Rounding using toward-zero
 -1.5 --> -1
  1.5 --> 1
Rounding using upward
 -1.5 --> -1
  1.5 --> 2
```

So some people expect this in some situations. Unfortunately setting a mode works very bad, from ITK build doesn't work for me in test. In particular case, IMHO, simple std::round (always to-nearest) may be OK. But it might be out of scope of the PR.

Reply:
> Unfortunately setting a mode works very bad,

Reproduced, cpp is from the above post, s. **-O3** flag. Probably this or similar was a reason to implement own _Round_.

```
r@deb2:~/tmp$ g++ rounding.cpp 
r@deb2:~/tmp$ ./a.out 

Rounding using to-nearest
 -1.5 --> -2
  1.5 --> 2
Rounding using toward-zero
 -1.5 --> -1
  1.5 --> 1
Rounding using upward
 -1.5 --> -1
  1.5 --> 2
```

```
r@deb2:~/tmp$ g++ -O3 rounding.cpp 
r@deb2:~/tmp$ ./a.out 

Rounding using to-nearest
 -1.5 --> -2
  1.5 --> 2
Rounding using toward-zero
 -1.5 --> -2
  1.5 --> 2
Rounding using upward
 -1.5 --> -2
  1.5 --> 2
```

Reply:
> I don't think people would _expect_ negative measurement values of the form `-n.5` (for a natural number `n`) to be rounded towards zero. Do you?

Similarly I don't think most people would expect a bias when rounding around zero and with other numeric computations or bining around zero. Here is a figure:
https://upload.cppreference.com/mwiki/images/7/7c/math-round_away_zero.svg